src/Resolvers/Microsoft.DotNet.MSBuildSdkResolver/Microsoft.DotNet.MSBuildSdkResolver.csproj

```diff
@@ -89,7 +89,7 @@
   <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'">
     <!-- Netfx version builds against the lowest version of System.Text.Json that's guaranteed to be shipped with MSBuild in VS -->
     <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" VersionOverride="8.0.0" />
-    <PackageReference Include="System.Text.Json" VersionOverride="8.0.0" />
```

Comment:
Wasn't this deliberately stale so that it'd work in one-release-old `MSBuild.exe`/VS? I don't think I understand why it has to be updated here.

@marcpopMSFT I remember discussing this with you last week sometime?

Reply:
Yes, this should not be updated until VS is updated to 8.0.4 version of STJ. Has that been done yet for 17.12?

For template locator and msbuild sdk resovler, we use the STJ version that VS ships so cannot move ahead of it.

CC @ViktorHofer @Forgind

Reply:
Without this it failed with
> src\Resolvers\Microsoft.DotNet.MSBuildSdkResolver\Microsoft.DotNet.MSBuildSdkResolver.csproj(160,5): error : (NETCORE_ENGINEERING_TELEMETRY=Build) Microsoft.DotNet.MSBuildSdkResolver is expected to depend on System.Text.Json, Version=8.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51. This may have consequences for MSBuild.exe binding redirects, please get signoff from the MSBuild team.

and since upgrading STJ was discussed in https://github.com/dotnet/msbuild/pull/10402 I thought this would be fine, sorry

Reply:
I think that's because ViktorHofer upgraded the STJ version in a few other places.

Also:
> This may have consequences for MSBuild.exe binding redirects, please get signoff from the MSBuild team.

🙂

Reply:
I saw that but I thought the msbuild discussion was that signoff 😄

Reply:
Submitted https://github.com/dotnet/deployment-tools/pull/381 to downgrade deployment-tools back to 8.0.3.